lib/attachments/gridstore-storage.js

```diff
@@ -129,6 +130,35 @@ class GridstoreStorage {
         let storeLock;
 
         let attachmentCallback = (...args) => {
+            // store finished uploading, add the hash of the file contents to file metadata
+            if (args.length > 2) {
+                const calculatedFileContentHash = args[2];
+
+                this.gridfs.collection(this.bucketName + '.files').findOneAndUpdate(
+                    {
+                        _id: hash
+                    },
+                    {
+                        $set: {
+                            'metadata.fileContentHash': calculatedFileContentHash
+                        }
+                    },
+                    {
+                        returnDocument: 'after'
+                    },
+                    (err, res) => {
+                        if (err) {
+                            return attachmentCallback(err);
```

Comment:
This will invoke `attachmentCallback()` and thus `callback()` twice. Make sure that `callback` is not run until the hash has been updated.